src/hotspot/share/runtime/safepoint.cpp

```diff
@@ -943,19 +943,21 @@ void ThreadSafepointState::print_on(outputStream *st) const {
 
 // Process pending operation.
 void ThreadSafepointState::handle_polling_page_exception() {
+  JavaThread* self = thread();
+  assert(self == Thread::current()->as_Java_thread(), "wrong thread");
```

Comment:
Perhaps: s/wrong thread/must be self/  or  s/wrong thread/must be calling thread/